src/approaches/grammar_search_invention_approach.py

```diff
@@ -676,6 +676,10 @@ def _evaluate_with_operators(self, predicates: FrozenSet[Predicate],
         seen_nondemos = 0
         max_demos = CFG.grammar_search_heuristic_based_max_demos
         max_nondemos = CFG.grammar_search_heuristic_based_max_nondemos
+        demo_atom_sets = {
```

Comment:
i don't get what this is... what is `a`? is this just a set of all abstract states seen in the demos?

Reply:
yeah